core/layer_manager.ts

```diff
@@ -122,7 +122,12 @@ export class LayerManager {
     if (!this.layers.has(layerNum)) {
       this.createLayer(layerNum);
     }
-    this.layers.get(layerNum)?.appendChild(elem.getSvgRoot());
+    const childElem = elem.getSvgRoot();
+    if (this.layers.get(layerNum)?.lastChild !== childElem) {
```

Comment:
This seems weird, since `append` doesn't remove the element from elsewhere, I can't imagine when this case would occur. Wouldn't it cause a problem if you tried to `append` a child to a layer where that child already existed?

If it is somehow intentional before that you could have the same element appended multiple times, it feels like you might be breaking things by disallowing that.

I don't have a good understanding of this code so I could be wrong, but this feels suspicious.

Reply:
See the first note in https://developer.mozilla.org/en-US/docs/Web/API/Node/appendChild which is actually what was happening. We would append the child which would trigger another focus, that would trigger an `onNodeFocus` callback for `RenderedWorkspaceComment`, which would append the child, and thus repeat ad infinitum.

Reply:
Oh got it, thanks!